src/taosdump.c

```diff
@@ -7327,12 +7372,22 @@ static int64_t dumpInAvroDataImpl(
                         break;
                     case TSDB_DATA_TYPE_JSON:
                     case TSDB_DATA_TYPE_NCHAR:
-                        if (field->type != TSDB_DATA_TYPE_NCHAR) {
-                            warnPrint("field[%d] type is not nchar/json!\n", i);
-                            bind->is_null = &is_null;
+                        // RecordSchema bytes only covert to nchar type
+                        if (field->type == TSDB_DATA_TYPE_NCHAR) {
+                            dumpInAvroDataNChar(field, &field_value, bind, &is_null);    
                         } else {
-                            dumpInAvroDataNChar(field, &field_value,
-                                    bind, &is_null);
+                            warnPrint("field[%d] type is not nchar/json! field->type=%d\n", i, field->type);
+                            bind->is_null = &is_null;                            
+                        }
+                        break;
+                    case TSDB_DATA_TYPE_VARBINARY:
+                    case TSDB_DATA_TYPE_GEOMETRY:
+                        // RecordSchema bytes only covert to nchar type
+                        if (field->type == TSDB_DATA_TYPE_NCHAR) {
```

Comment:
TSDB_DATA_TYPE_NCHAR is typo in case varbinary/geometry?

Reply:
field->type is come from avro bytes type, map to TDengine's nchar type, so only have one type, may varbinary or geometry, so prompt this warning is ok.